### DISCUSSION TITLE
Garage Door JSON

### DISCUSSION THREAD

**Discussion by nickjmassa** [RavenSystem/esp-homekit-devices#1592] [Q&A] [answered]
Hi, Trying to get off the ground here. fullhaaboot.bin was flashed to a HiLetGo ESP8266 ESP12E via NodeMCU PyFlasher. I can reach the local web page, have set up Wifi, and input the below JSON string. However I cannot get it to show up in the Apple Home app (I've tried the QR and manual code entry). I've also reset the HomeKit ID, and unplugged it and plugged it back in - not sure what I'm doing wrong here? General setup is relay attached to GPIO5 (aka D1) and reed switch attached to GPIO4 (aka D2).

`{“c”:{”n”:“haagarage”},”a”:[{“t”:40,"d":18,“0”:{“r”:[{“g”:5,”v”:1,”i”:0.5}]},”1”:{“a”:0},“2”:{“r”:[{“g”:5,”v”:1,”i”:0.5},{“g”:5,”i”:1.5},{“g”:5,”v”:1,”i”:0.5}]},”3”:{“a”:2},”f3":[{"g":4,"t":0}],"f4":[{"g":4,”t”:1}]}]}`

Thanks! - Nick

-Edit-
Got to pull up a log file and it said invalid JSON. Tested the exact example from the garage door example and it worked. I think this has to do with what pins I am using. Is there a limit on what GPIO pins can be used? I plan on using the following GPIOs(D# pins) for 3 relays and 3 reed switches for my garage doors - 5(D1), 4(D2), 14(D5), 12(D6), 13(D7), and (output only) 0(D3).

![67606867-5b8d1c00-f78b-11e9-84c5-7a78ec614328](https://user-images.githubusercontent.com/43593508/138978191-8b778a3e-63c2-4899-8f58-38cf0e10540f.jpg)

**Comment by tonysprenk** [comment, marked as answer]
I’m guessing you copied the json to the notes app and from the notes app you copied to the device. Sometimes the notes app changes the style of quotation marks. In the json you copied above the wrong style of quotation marks are present. It should be: " , not: “ . Hope this helps.

**Reply by nickjmassa** [reply]
🤦‍♂️ installed atom editor and now fully up and running. Thanks Tony!
